tests/upgrades/test_remoteexecution.py

```diff
@@ -17,103 +17,114 @@
 
         Test Steps:
 
         1. Before Satellite upgrade:
-           a. Create Content host.
-           b. Create a Subnet on Satellite.
-           c. Install katello-ca on content host.
-           d. Register content host to Satellite.
-           e. add_ssh_key of external capsule on content host.
-           f. run a REX job on content host.
+           a. Register content host to Capsule.
+           b. run a REX job on content host.
         2. Upgrade satellite/capsule.
-        3. Run a rex Job again with same content host.
+        3. Run a REX Job again with same content host.
         4. Check if REX job still getting success.
     """
 
-    @pytest.mark.pre_upgrade
+    @pytest.mark.rhel_ver_list([8])
     @pytest.mark.no_containers
-    def test_pre_scenario_remoteexecution_external_capsule(
-        self, request, default_location, rhel7_contenthost, save_test_data
+    @pytest.mark.pre_upgrade
+    def test_pre_scenario_remote_execution_external_capsule(
+        self,
+        rhel_contenthost,
+        target_sat,
+        module_org,
+        smart_proxy_location,
+        module_ak_with_cv,
+        save_test_data,
     ):
-        """Run REX job on client registered with external capsule
+        """
+        Run REX job on client registered with external capsule
 
         :id: preupgrade-261dd2aa-be01-4c34-b877-54b8ee346561
 
         :steps:
-            1. Create Subnet.
-            2. Create Content host.
-            3. Install katello-ca package and register to Satellite host.
-            4. add rex ssh_key of external capsule on content host.
-            5. run the REX job on client vm.
+            1. Create Content host.
+            2. add rex ssh_key of external capsule on content host.
+            3. run the REX job on client vm.
 
         :expectedresults:
             1. Content host should create with pre-required details.
             2. REX job should run on it.
 
         :parametrized: yes
+
         """
-        sn = entities.Subnet(
-            domain=self.vm_domain,
-            gateway=self.gateway,
-            ipam='DHCP',
-            location=[default_location.id],
-            mask=self.netmask,
-            network=self.subnet,
-            organization=[self.org.id],
-            remote_execution_proxy=[entities.SmartProxy(id=2)],
-        ).create()
-        rhel7_contenthost.install_capsule_katello_ca(capsule=self.proxy_name)
-        rhel7_contenthost.register_contenthost(org=self.org.label, lce='Library')
-        capsule = Capsule(self.proxy_name)
-        rhel7_contenthost.add_rex_key(satellite=capsule)
-        host = entities.Host().search(query={'search': f'name="{rhel7_contenthost.hostname}"'})
-        host[0].subnet = sn
-        host[0].update(['subnet'])
-        job = entities.JobInvocation().run(
+        rhel_contenthost._skip_context_checkin = True
+        capsule = Capsule(target_sat.api.SmartProxy(id=2).read().name)
```

Comment:
For now, I think getting the external capsule with `id=2` is fine and we will switch to dynamic when we will spin up capsule on the fly.

Just recording here for the purpose.